File: frontend/sistema-academico/src/app/shared/dto/diaHora.dto.ts
import { DisciplinaOferecidaDto } from './disciplina-oferecida.dto';

export interface DiaHoraDto {
    diaHoraId?: number;
    horario?: string;
    diaSemana?: string;
    disciplinaOferecida?: DisciplinaOferecidaDto;
  }

File: backend/digitalhouse-pi/src/main/java/com/pi/dh/dto/DiaHoraDto.java
package com.pi.dh.dto;

import java.util.List;

import com.pi.dh.domain.model.Aluno;
import com.pi.dh.domain.model.enums.DiaSemana;
import com.pi.dh.domain.model.enums.Hora;

import lombok.Data;

@Data
public class DiaHoraDto {
	
	private Long diaHoraId;
	private Hora horario;
	private DiaSemana diaSemana;
	private List<Aluno> alunos;
	
	public String getHorario() {
    	return horario.getHora();
    }
    
    public String getDiaSemana() {
    	return diaSemana.getDia();
    }

}
